yetanotherco/aligned_layer
# chore(aggregator): implement telemetry for aggregator

# Description
This PR allows the aggregator to export telemetry traces to have a better observability of the system

# Tasks

- [x] Docker compose with Otel Collector and Jaeger UI
- [x] Init traces on NewTask
- [x] Add span on Operator responses
- [x] Add span on Quorum Reached
- [x] Add span on Tasks errors (ie task expired)
- [x] Close traces after a period of time the quorum was reached to wait for most operators responses 
    

# How to Test

> [!NOTE]  
> Docker is needed to test the PR.

1. Run telemetry
```
make run_telemetry
```
It will run the Otel-Collector and Jaeger UI.
> [!NOTE]  
> Jaeger UI is available on http:localhost:16686
<img width="1904" alt="image" src="https://github.com/user-attachments/assets/91713ca6-b26a-4443-9543-c358f4452591">

2. Run anvil
```
make anvil_start_with_block_time
```

3. Run aggregator
```
make aggregator_start
```

4. Run operator
```
export OPERATOR_ADDRESS=0x70997970C51812dc3A010C7d01b50e0d17dc79C8
make operator_full_registration CONFIG_FILE=config-files/config-operator-1.yaml
make operator_start CONFIG_FILE=config-files/config-operator-1.yaml
```

5. Run batcher
```
make batcher_start_local
```

6. Send tasks 

7. Once you sent a batch and it has been responded by the aggregator, you should be able to check the trace of the task in the Jaeger UI (`http:localhost:16686`)

In the search, select service: `aggregator` and press `find traces`
It should show you the tasks you sent to aligned
